### DISCUSSION TITLE
Licensing of generated Project

### DISCUSSION THREAD

**Discussion by TheGreatRefrigerator** [fastapi/fastapi#8467] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# no code
```

### Description

This question is referring to this [issue in the project generator](https://github.com/tiangolo/full-stack-fastapi-postgresql/issues/457).
As there is not much activity there but the project generator is linked to from the official [fastapi documentation](https://fastapi.tiangolo.com/project-generation) and also recommended to be used, this might still be relevant here.

**The license of the project generator is MIT, does this still apply for the generated project?**

I would like to use the AGPL license for my project but am not sure if i can just do so.

Preferably i would just have the github readme state that it is based the project on the generator and not have to specifically state every change made after the base generation of the project (which some sources i read suggest).

### Operating System

macOS

### FastAPI Version

0.75.2

### Python Version

3.9.7

**Comment by jgould22** [comment, marked as answer]
Licenses are very tricky things.

Additionally I am not a lawyer.

That said, MIT is very permissive license. As such since the generator is licensed MIT you would probably be fine to fork it and add the License field to the `cookiecutter.json` to add the license of your choice and generate it under that license using cookie cutter.

**Reply by d4vidsha** [reply]
Is `cookiecutter.json` still a viable option? I can't find it in the repository. I also want to change MIT license to AGPLv3 but not sure what I can do to achieve this.

**Reply by jgould22** [reply]
Looks like they have updated the repo and no longer use cookiecutter to generate a project.

Copier is now offered instead of cookie cutter, you can edit the copier config to put a license field as seen here https://copier.readthedocs.io/en/stable/configuring/
